modin/core/dataframe/algebra/groupby.py

```diff
@@ -121,6 +121,7 @@ def map(
         # right index and placing columns in the correct order.
         groupby_args["as_index"] = True
         groupby_args["observed"] = True
+        apply_func = cls.try_filter_dict(map_func, df)
```

Comment:
please add this info as a comment, something like
```python
# cls.try_filter_dict() should go before insertion of missing 'by' columns, otherwise
# if 'by' is presented in the agg dict, the aggregation of 'by' cols would be computed in every partition
```

Reply:
Done, added the following in-code comment:
```python
        # We have to filter func-dict BEFORE inserting broadcasted 'by' columns
        # to avoid multiple aggregation results for 'by' cols in case they're
        # present in the func-dict:
```